frontend/packages/console-app/src/components/cloud-shell/CloudShellExec.tsx

```diff
@@ -0,0 +1,159 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { Base64 } from 'js-base64';
+import { LoadingBox, StatusBox } from '@console/internal/components/utils';
+import { connectToFlags, WithFlagsProps } from '@console/internal/reducers/features';
+import { impersonateStateToProps } from '@console/internal/reducers/ui';
+import { FLAGS } from '@console/shared';
+import { WSFactory } from '@console/internal/module/ws-factory';
+import { resourceURL } from '@console/internal/module/k8s';
+import { PodModel } from '@console/internal/models';
+import { Terminal } from './AutoResizeTerminal';
+
+// pod exec WS protocol is FD prefixed, base64 encoded data (sometimes json stringified)
+
+// Channel 0 is STDIN, 1 is STDOUT, 2 is STDERR (if TTY is not requested), and 3 is a special error channel - 4 is C&C
+// The server only reads from STDIN, writes to the other three.
+// see also: https://github.com/kubernetes/kubernetes/pull/13885
+
+type CloudShellExecProps = {
+  container: string;
+  podname: string;
+  namespace: string;
+  shcommand?: string[];
+  message?: string;
+  shouldResize?: boolean;
+};
+
+type CloudShellExecState = {
+  open: boolean;
+  error: string;
+};
+
+const NO_SH =
+  'starting container process caused "exec: \\"sh\\": executable file not found in $PATH"';
+
+class CloudShellExec extends React.PureComponent<
+  CloudShellExecProps & StateProps & WithFlagsProps,
+  CloudShellExecState
+> {
+  private terminal;
+
+  private ws;
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: false,
+      error: null,
+    };
+    this.terminal = React.createRef();
+  }
+
+  connect = () => {
+    const { container, podname, namespace, shcommand, flags, impersonate } = this.props;
+    const usedClient = flags[FLAGS.OPENSHIFT] ? 'oc' : 'kubectl';
+    const cmd = shcommand || ['sh', '-i', '-c', 'TERM=xterm sh'];
+
+    const params = {
+      ns: namespace,
+      name: podname,
+      path: 'exec',
+      queryParams: {
+        stdout: '1',
+        stdin: '1',
+        stderr: '1',
+        tty: '1',
+        container,
+        command: cmd.map((c) => encodeURIComponent(c)).join('&command='),
+      },
+    };
+
+    if (this.ws) {
+      this.ws.destroy();
+      const { current } = this.terminal;
```

Comment:
using `current` is a bad name and makes following the code harder as the variable name isn't descriptive enough. This is a general comment for other areas that follow the same pattern.